okio/src/commonMain/kotlin/okio/FileSystem.kt

```diff
@@ -376,6 +378,18 @@ expect abstract class FileSystem() {
   @Throws(IOException::class)
   abstract fun createSymlink(source: Path, target: Path)
 
+  /**
+   * Returns a new file system that forwards all calls to this, and that also returns [extension]
+   * when it is requested.
+   *
+   * When [extensionType] is requested on the returned file system, it will return [extension],
+   * regardless of what is returned by this file system.
+   */
+  open fun <E : FileSystemExtension> extend(extensionType: KClass<E>, extension: E): FileSystem
```

Comment:
Should there be a way to freeze these? And reopen via a new builder() type pattern.

Otherwise FileSystem.System is at risk of monkey patching.

Reply:
Not an issue since it returns a new one.